### ISSUE TITLE
Add instances for NonEmpty

### ISSUE THREAD

**Issue by phadej** [GetShopTV/swagger2#93]

**Comment by lucasdicioccio** [comment]
Started something at https://github.com/lucasdicioccio/swagger2/commit/d748ab74a444e2af092f3dc0263bc9a10e4f644f . I'll take some more time to actually test the thinggy and try whether GHC7 works with the same instance.

(currently using something along these lines at work on a newtype defined over NonEmpty).

**Comment by lucasdicioccio** [comment]
Obviously I mixed ToSchema and ToParamSchema in this quick fix before leaving :-). I'll do a proper one later.

**Comment by lucasdicioccio** [comment]
Ok, updated 
my fix https://github.com/lucasdicioccio/swagger2/commit/a1dbad9b34282d914146caeb751c2a13b09f2e31 .

Do you want to discuss over the Pull Request or in this thread? In particular, should I add an instance for GHC<8?

**Comment by phadej** [comment]
please make a PR, and yes, please add a `semigroups` dependency, as 

```
  if !impl(ghc >= 8.0)
    build-depends:
        semigroups  >= 0.18.3 && <0.19
```

(latest `semigroups` is `0.18.3` atm, no need to push lower bound down, as we won't test that anyway)

**Comment by phadej** [comment]
please make a PR, and yes, please add a `semigroups` dependency, as 

```
  if !impl(ghc >= 8.0)
    build-depends:
        semigroups  >= 0.18.3 && <0.19
```

(latest `semigroups` is `0.18.3` atm, no need to push lower bound down, as we won't test that anyway)

**Comment by fizruk** [comment]
Closed in 22e975e.
